msticpy/context/azure/sentinel_ti.py

```diff
@@ -0,0 +1,455 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""Mixin Classes for Sentinel Analytics Features."""
+from datetime import datetime
+from typing import Optional
+
+import httpx
+import pandas as pd
+from azure.common.exceptions import CloudError
+
+from ..._version import VERSION
+from ...common.exceptions import MsticpyUserError
+from .azure_data import get_api_headers
+from .sentinel_utils import _azs_api_result_to_df, _build_sent_data, get_http_timeout
+
+__version__ = VERSION
+__author__ = "Pete Bryan"
+
+
+_INDICATOR_ITEMS = {
+    "valid_to": "validUntil",
+    "description": "description",
+    "threat_types": "threatTypes",
+    "name": "displayName",
+    "confidence": "confidence",
+    "revoked": "revoked",
+    "source": "source",
+}
+
+
+class SentinelTIMixin:
+    """Mixin class for Sentinel Hunting feature integrations."""
+
+    def get_all_indicators(
+        self,
+        limit: Optional[int] = None,
+        orderby: Optional[str] = None,
+    ) -> pd.DataFrame:
+        """
+        Return all TI indicators in a Microsoft Sentinel workspace.
+
+        Parameters
+        ----------
+        limit: int, optional
+            If set returns top n results
+        orderby : Optional[str], optional
+            Order results by a specific column
+
+        Returns
+        -------
+        pd.DataFrame
+            A table of the custom hunting queries.
+
+        """
+        appendix = "/indicators?"
+        if limit:
+            appendix += f"&$top={limit}"
+        if orderby:
+            appendix += f"&$orderby={orderby}"
+        return self._list_items(  # type: ignore
+            item_type="ti_path", api_version="2021-10-01", appendix=appendix
+        )  # type: ignore
+
+    def get_ti_metrics(self) -> pd.DataFrame:
+        """
+        Return metrics about TI indicators in a Microsoft Sentinel workspace.
+
+        Returns
+        -------
+        pd.DataFrame
+            A table of the custom hunting queries.
+
+        """
+        return self._list_items(  # type: ignore
+            item_type="ti_path", api_version="2021-10-01", appendix="/metrics"
+        )  # type: ignore
+
+    def create_indicator(
+        self,
+        indicator: str,
+        ioc_type: str,
+        name: str = "TI Indicator",
+        confidence: int = 0,
+        silent: bool = False,
+        **kwargs,
+    ):
+        """
+        Create a new indicator within the Microsoft Sentinel workspace.
+
+        Parameters
+        ----------
+        indicator : str
+            The indicator to create - i.e. IP address, domain name etc.
+        ioc_type : str
```

Comment:
Wondering if we should allow the the IoC types that we use as aliases?
dns, file_hash, ipv4, ipv6?

This is the ioc enum from iocextract.py
```python
    ipv4 = "ipv4"
    ipv6 = "ipv6"
    dns = "dns"
    url = "url"
    md5_hash = "md5_hash"
    sha1_hash = "sha1_hash"
    sha256_hash = "sha256_hash"
    file_hash = "file_hash"
    email = "email"
    windows_path = "windows_path"
    linux_path = "linux_path"
    hostname = "hostname"
```

Also, this list doesn't match the list you check against in the body of the function.
(if we allowed file_hash, we'd have to check length 32 (md5), 40 (sha1), 64 (sha256) and map it to the appropriate type)

Reply:
We can't use all of these as a lot aren't supported by the API. But good idea on aligning. Updated to map across the two